File: members/U02BQ6G1SQJ/programmers/problem/징검다리 건너기.py
(MIN_STONE, MAX_STONE) = (1, 200000000);

def checkPossible(stoneList : list, maxJump : int, candidateAnswer : int) -> bool:
    # print(stoneList);
    
    zeroCnt = 0;
    
    for curStone in stoneList:
        if (curStone > 0):
            zeroCnt = 0;
        else:
            zeroCnt += 1;
            
            if (zeroCnt == maxJump):
                return False;
    
    return True;

def solution(stones : list, k : int) -> int:
    (answer, left, right) = (0, MIN_STONE, MAX_STONE);
    
    while (left <= right):
        mid = (left + right) // 2;
        
        # print(mid, left, right);
        
        if (checkPossible(list(map(lambda k : k - mid, stones)), k, mid)):
            left = mid + 1;
        else:
            (answer, right) = (mid, mid - 1);
    
    return answer;

File: members/U02CDRR78MN/kakaoIntern2019/징검다리건너기.js
function solution(stones, k) {
  let left = 1;
  let right = 200000000;

  const possible = (num) => {
    let dangerous = 0;

    for (const stone of stones) {
      if (stone >= num) {
        dangerous = 0;
      } else {
        dangerous += 1;
      }

      if (dangerous === k) {
        return false;
      }
    }
    return true;
  };

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);

    if (possible(mid)) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return right;
}
